### PR TITLE
(proof-new) Add new arithmetic kind INDEXED_ROOT_PREDICATE.

### PR DESCRIPTION
**Pull request by nafur**
This PR introduces a new arithmetic kind for indexed root predicates.
An indexed root predicate compares a real variable to the k'th root of a given polynomial as follows:
Let `IRP_k(x ~ 0, p)` an indexed root predicate with `k` a non-negative number, `x` some real variable, `~` an arithmetic relation (e.g. `=`, `<`, ...), and `p` a polynomial over `x` (and possibly other variables).
If `p` contains variables apart from `x`, we can only evaluate the expression over a suitable assignment for at least these variables.
The evaluation of this expression is equivalent to computing the k'th real root of `p` in `x` (with all other variables evaluated over a given assignment) and comparing this real root to zero (according to the relation symbol `~`).

Note that we currently do not intend to use this structure for solving, but require it for representing and printing CAD proofs.

### CODE REVIEW

**Comment by nafur** [comment]
I feel this PR should add some proper documentation on what this kind means (similar to what I explained in the comment of the PR), but I'm not sure where this should go...

File: src/util/arith.h [hunk 1]
```diff
@@ -0,0 +1,58 @@
+/*********************                                                        */
```
Comment (ajreynol) [inline src/util/arith.h:1]: Should we just call this file index_root_predicate.h?

File: src/util/arith.h [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+/*! \file arith.h
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Aina Niemetz, Andres Noetzli, Dejan Jovanovic
```
Comment (ajreynol) [inline src/util/arith.h:5]: Update.

File: src/util/arith.h (hunk 1 above)
Comment (nafur) [reply src/util/arith.h:1]: Yes, we should probably do that.

**Review by ajreynol — APPROVED** [review]

**Pull request merged**